pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -27,19 +29,18 @@ import (
 )
 
 func newETCD3Storage(c storagebackend.Config) (storage.Interface, DestroyFunc, error) {
-	tlsInfo := transport.TLSInfo{
-		CertFile: c.CertFile,
-		KeyFile:  c.KeyFile,
-		CAFile:   c.CAFile,
-	}
-	tlsConfig, err := tlsInfo.ClientConfig()
-	if err != nil {
-		return nil, nil, err
-	}
-	// NOTE: Client relies on nil tlsConfig
-	// for non-secure connections, update the implicit variable
-	if len(c.CertFile) == 0 && len(c.KeyFile) == 0 && len(c.CAFile) == 0 {
-		tlsConfig = nil
+	var tlsConfig *tls.Config
+	if !(len(c.CertFile) == 0 && len(c.KeyFile) == 0 && len(c.CAFile) == 0) {
```

Comment:
i am not sure if this is better. previously, get config would complain if user pass in a partial config. now we simply ignore partial config and use non-tls.